core/src/main/java/bisq/core/xmr/connection/MoneroConnectionManager.java

```diff
@@ -0,0 +1,144 @@
+package bisq.core.xmr.connection;
+
+import bisq.core.xmr.connection.model.MoneroConnection;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.time.Duration;
+
+import java.net.URI;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+
+
+import monero.common.MoneroRpcConnection;
+
+@Slf4j
+@Singleton
+public class MoneroConnectionManager {
+
+    private final Object lock = new Object();
```

Comment:
Lock should not be necessary? monero-java's `MoneroConnectionManager` should be made thread safe if necessary.

Reply:
Depends if `MoneroConnectionsManager` can be used by multiple threads. Then the synchronization will keep the store and the `MoneroConnectionManager` in synch and is actually required. If it is not used by multiple threads, then the synchronization can be removed